File: Matrices/Medium/48/48.py
from typing import List

class Solution:
    def rotate(matrix: List[List[int]]) -> None:
        """
        Do not return anything, modify matrix in-place instead.
        """
        n = len(matrix)
        for i in range(n):
            for j in range(i+1, len(matrix[i])):
                matrix[i][j], matrix[j][i] = matrix[j][i], matrix[i][j]

        for i in range(n):
            # for j in range(len(matrix[i]) // 2):
            #     matrix[i][j], matrix[i][n-j-1] = matrix[i][n-j-1], matrix[i][j]
            matrix[i].reverse()

File: Matrices/Medium/48/48.c
void rotate(int** matrix, int matrixSize, int* matrixColSize){
    for(int i = 0; i < matrixSize; i++){
        for(int j = i+1; j < matrixSize; j++){
            int temp = matrix[i][j];
            matrix[i][j] = matrix[j][i];
            matrix[j][i] = temp;
        }
    }
    for(int i = 0; i < matrixSize; i++){
        for(int j = 0; j < matrixSize/2; j++){
            int temp = matrix[i][j];
            matrix[i][j] = matrix[i][matrixSize-j-1];
            matrix[i][matrixSize-j-1] = temp;
        }
    }
}